src/compiler/utilities.ts

```diff
@@ -3465,8 +3465,9 @@ namespace ts {
     }
 
     function getReplacement(c: string, offset: number, input: string) {
-        if (c.charCodeAt(0) === CharacterCodes.nullCharacter) {
-            const lookAhead = input.charCodeAt(offset + c.length);
+        const charCode = c.charCodeAt(0);

```

Comment:
this extract variable doesn't seem necessary to me

Reply:
You’re right, I had initially made other changes in this function.